pykokkos/core/cpp_setup.py

```diff
@@ -323,10 +328,14 @@ def get_cuda_compute_capability(self, compiler: str) -> str:
     @staticmethod
     def is_compiled(output_dir: Path) -> bool:
         """
-        Check if an entity is compiled
+        Check if an entity is compiled. It does this via checking if any .so file exists.
 
         :param output_dir: the directory containing the compiled entity
-        :returns: true if compiled
+        :returns: true if any .so object exists
         """
+        so_files = output_dir.glob("*.so")
+        contains_so_files = False
+        for file in so_files:
+            contains_so_files = file.is_file()
```

Comment:
This now checks for ANY .so file in the path. Timing with a 1000 iterations shows, that asking if the dir exists has the same cost as checking for the .so file (if only one is there). This also triggers a recompile if the .so was not found. And if the recompile fails gives the instructions for the colored compiler output.

Nevertheless, we do not check for the name, as this would require a larger refactor (the method could no longer be static or we would need to pass in the module name from outside). For the different backends we should be fine, as the backend name is contained in the folder path. But we might still need the module names, as we have this multigpu thing where multiple .so files with different names exist in the same folder. But someone first using only one gpu and then multiple without clearing the cache might be an edge case we do not care about...